index.js

```diff
@@ -23,59 +23,63 @@ function WebSocketWrapper(url) {
 	Emitter.call(this);
 
 	this.url = url;
-	this._socket = undefined;
-	this._connect();
+	this._socket = this._connect();
 }
 WebSocketWrapper.prototype = Object.create(Emitter.prototype);
 
-// wrap the states
-WebSocketWrapper.prototype.CONNECTING = 0;
-WebSocketWrapper.prototype.OPEN = 1;
-WebSocketWrapper.prototype.CLOSING = 2;
-WebSocketWrapper.prototype.CLOSED = 3;
+WebSocketWrapper.CONNECTING = 0;
+WebSocketWrapper.OPEN = 1;
+WebSocketWrapper.CLOSING = 2;
+WebSocketWrapper.CLOSED = 3;
 
 // wrap readyState
 Object.defineProperty(WebSocketWrapper.prototype, 'readyState', {
 	enumerable: true,
 	get: function () {
-		return this._socket ? this._socket.readyState : this.CLOSED;
+		return this._socket ? this._socket.readyState : WebSocketWrapper.CLOSED;
 	}
 });
 
 // wrap this for easier extensibility
 WebSocketWrapper.prototype.send = function WebSocketWrapper_send(msg) {
-	if (this.readyState === this.OPEN)
-		this._send(msg);
+	if (this.readyState === WebSocketWrapper.OPEN) {
+		// Calling send() on a closed WebSocket object which reports an open
+		// readyState causes a crash. This scenario can occur when returning to
+		// a backgrounded page which received data and then closed when in the
+		// backgrounded state.
+		// https://gist.github.com/mloughran/2052006
+		setTimeout(function () {
```

Comment:
is setting a time out the only option to fix this? or just a temporary solution to check if this is the problem?